compiler/src/main/scala/org/scalajs/core/compiler/JSEncoding.scala

```diff
@@ -103,6 +103,8 @@ trait JSEncoding extends SubComponent { self: GenJSCode =>
     val idSuffix =
       if (sym.isPrivate || allRefClasses.contains(sym.owner))
         sym.owner.ancestors.count(!_.isTraitOrInterface).toString
+      else if (sym.isOuterField)
+        sym.owner.name
```

Comment:
Weee... class overrides anyone?

For non-french speakers: @sjrd says this will need *very* serious binary compat investigation.